src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkConnectorConfig.java

```diff
@@ -373,6 +380,10 @@ public boolean dataStreamEnabled() {
         return getBoolean(DATA_STREAM_ENABLED);
     }
 
+    public String dataStreamExistingIndexTemplateName() {
```

Comment:
Please:

```suggestion
    public Optional<String> dataStreamExistingIndexTemplateName() {
```

We could also make sure if the template name is empty string, `Option.empty` is returned

Reply:
Indeed, it looks better and aligns with other fields. Updated.